cmd/kms-handlers.go

```diff
@@ -707,15 +688,20 @@ func (a kmsAPIHandlers) KMSListIdentitiesHandler(w http.ResponseWriter, r *http.
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrNotImplemented), r.URL)
 		return
 	}
-	identities, err := manager.ListIdentities(ctx, r.Form.Get("pattern"))
+	identities, err := manager.ListIdentities(ctx)
 	if err != nil {
 		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 		return
 	}
-	values, err := identities.Values(0)
-	if err != nil {
-		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
-		return
+	var values []kes.IdentityInfo
+	for name, err := identities.SeekTo(ctx, r.Form.Get("pattern")+"*"); err != io.EOF; name, err = identities.Next(ctx) {
```

Comment:
Do you want to add * only if the pattern itself does not have a * ?

Reply:
👍